test/libsolidity/SemanticTest.cpp

```diff
@@ -331,11 +359,28 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 		return;
 
 	_stream << _linePrefix << "// ====" << endl;
-	if (m_compileViaYulCanBeSet)
+	if (m_compileToEwasmCanBeSet)
+	{
+		// if test was already configured to run via yul,
+		// we need to preserve the original settings.
+		if (m_runWithYul)
+		{
+			// if test was also configured to run without yul.
+			if (m_runWithoutYul)
+				_stream << _linePrefix << "// compileViaYul: also\n";
+			// if test was configured only to run with yul.
+			else
+				_stream << _linePrefix << "// compileViaYul: true\n";
+		}
+
+		_stream << _linePrefix << "// compileToEwasm: also\n";
+	}
+	else if (m_compileViaYulCanBeSet)
 		_stream << _linePrefix << "// compileViaYul: also\n";
+
 	for (auto const& setting: settings)
-		if (!m_compileViaYulCanBeSet || setting.first != "compileViaYul")
-		_stream << _linePrefix << "// " << setting.first << ": " << setting.second << endl;
+		if (!(m_compileViaYulCanBeSet || m_compileToEwasmCanBeSet) || setting.first != "compileViaYul")
+			_stream << _linePrefix << "// " << setting.first << ": " << setting.second << endl;
 }
```

Comment:
I think that this code now swallows the `compileToWasm` flag when `m_compileToEwasmCanBeSet` is false but `m_compileViaYulCanBeSet` is true. And it also swallows `compileViaYul` flag when `m_compileViaYulCanBeSet` is false but `m_compileToEwasmCanBeSet` is true and the flag was originally false in the file.